File: #2466.count-ways-to-build-good-strings.cpp
class Solution {
public:
    int countGoodStrings(int low, int high, int zero, int one) {
        
        long long dp[100000 + 1] = {0};
        dp[zero]++;
        dp[one]++;
        int l = zero < one ? zero : one;
        for (int i = l + 1; i <= high; i++) {
            if ((i-zero ) > 0) {
                dp[i] += (dp[i - zero] % long(pow(10, 9) + 7));
            }

            if ((i - one) > 0) {
                 dp[i] += (dp[i - one] % long(pow(10, 9) + 7));
            }
        }

        long long sum = 0;
        for (int i = low; i <= high; i++) {
            sum += dp[i];
        }
        return sum % (int(pow(10, 9)) + 7);
    }
};

File: #2466.count-ways-to-build-good-strings.js
/**
 * @param {number} low
 * @param {number} high
 * @param {number} zero
 * @param {number} one
 * @return {number}
 */
var countGoodStrings = function(low, high, zero, one) {
    const MOD = 1e9 + 7;
    const DP = new Array(high + 1).fill(0);
    let ans = 0;
    DP[0] = 1;
    for (let i = 1; i <= high; i++) {
        DP[i] = ((i >= zero ? DP[i - zero] : 0)
            + (i >= one ? DP[i - one] : 0)) % MOD;
        if (i >= low && i <= high) {
            ans = (ans + DP[i]) % MOD;
        }
    }
    return ans;
};
